velox/docs/develop/operators.rst

```diff
@@ -214,6 +218,73 @@ aggregate functions.
 Finally, note that computing measures over sorted input is only possible if
 aggregation step is 'single'. Such computations cannot be split into partial + final.
 
+To illustrate the need for globalGroupingSets and groupIdColumn, we examine the following SQL
+
+.. code-block:: sql
+
+  SELECT orderkey, sum(total_quantity) FROM orders WHERE GROUP BY CUBE (orderkey);
+
+This is equivalent to the following SQL with GROUPING SETS
+
+.. code-block:: sql
+
+  SELECT orderkey, sum(total_quantity) FROM orders GROUP BY GROUPING SETS ((orderkey), ());
+
+The SQL gives sub-totals of total_quantity for each orderkey along with the global sum (from the empty grouping set).
+
+The optimizer plans the above query as a Aggregation over a GroupId node. The GroupId node duplicates
+the input rows for each grouping set.
+
+Lets say the orders table has 5 rows:
+
+.. code-block::
+
+orderkey   total_quantity
+   1            5
+   2            6
+   2            7
+   3            8
+   4            9
+
+After GroupId for the grouping sets ((orderkey), ()) the table has the following 10 rows
+
+.. code-block::
+
+orderkey   total_quantity    group_id
+   1            5                0
+   2            6                0
+   2            7                0
+   3            8                0
+   4            9                0
+   null         5                1
+   null         6                1
+   null         7                1
+   null         8                1
+   null         9                1
+
+A subsequent aggregation with grouping keys (orderkey, group_id) gives the sub-totals for the query
+
+.. code-block::
+
+orderkey   total_quantity
+  1           5
+  2           13
+  3           8
+  4           9
+  null        35
+
+If there were no input rows for this GROUP BY CUBE, then the expected result is a single row with the default value for the
+global aggregation. For the above query that would be:
+
+.. code-block::
+
+orderkey   total_quantity
+  null           null
+
+To generate this special row the AggregationNode needs the groupId for the global grouping set (1 in this case) and it
```

Comment:
Thank you for adding this explanation. It helps. Can you also add something to explain why globalGroupingSets (multiple), not globalGroupingSet (one)?